exercises/allergies/.gitignore

```diff
@@ -0,0 +1,7 @@
+# Generated by Cargo
+# will have compiled files and executables
+/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
```

Comment:
i left it in because i want it to simulate what someone might actually have to do if they create a project from scratch. if they make a project on GitHub, this is the auto-created `.gitignore`.

leaving the statement in there, 1) teaches students that the `.gitignore` should be different for executables vs libraries, 2) gives students the exercise of cleaning it up (similar to having to make the `src` directory and the `lib.rs` file), 3) does no harm if not modified.

so i see this as 2 learning opportunites vs 0 problems, so i'd leave it in.

that being said, if everyone disagrees with me i am happy to change the PR (this is not a mountain i'm going to die on 😆)